.github/workflows/test-linux.yml

```diff
@@ -0,0 +1,264 @@
+      - name: Prepare venv
+        run: |
+          set -e
+          python -m pip install --upgrade pip setuptools wheel
+          python -m venv test-job
+      - name: Install Qiskit from sdist
+        if: ${{ inputs.install-from-sdist }}
+        run: |
+          set -e
+          # Use stable Rust, rather than MSRV, to spot-check that stable builds properly.
+          rustup override set stable
+          source test-job/bin/activate
+          python -m pip install -U pip build
+          python -m build --sdist .
+          python -m pip install -U \
+            -c constraints.txt \
+            -r requirements.txt \
+            -r requirements-dev.txt \
+            dist/qiskit-*.tar.gz
+          # Build and install both qiskit and qiskit-terra so that any optionals
+          # depending on `qiskit` will resolve correctly.
+      - name: Install Qiskit directly
+        if: ${{ !inputs.install-from-sdist }}
+        run: |
+          set -e
+          source test-job/bin/activate
+          python -m pip install -U \
+            -c constraints.txt \
+            -r requirements.txt \
+            -r requirements-dev.txt \
+            -e .
+          # Build and install both qiskit and qiskit-terra so that any optionals
+          # depending on `qiskit` will resolve correctly.
+      - name: Install Optional packages
+        run: |
+          set -e
+          source test-job/bin/activate
+          python -m pip install -r requirements-optional.txt -c constraints.txt
+          python -m pip check
+        if: ${{ inputs.install-optionals }}
+      - name: Install optional non-Python dependencies
+        run: |
+          set -e
+          sudo apt-get update
+          sudo apt-get install -y graphviz
+        if: ${{ inputs.install-optionals }}
+      - name: Run Python tests
+        run: |
+          set -e
+          source test-job/bin/activate
+          python tools/report_numpy_state.py
+          mkdir -p /tmp/terra-tests
+          cp -r test /tmp/terra-tests/.
+          cp .stestr.conf /tmp/terra-tests/.
+          cp -r .stestr /tmp/terra-tests/. || :
+          pushd /tmp/terra-tests
+          export PYTHONHASHSEED=$(python -S -c "import random; print(random.randint(1, 4294967295))")
+          echo "PYTHONHASHSEED=$PYTHONHASHSEED"
+          stestr run
+          popd
+        env:
+          QISKIT_PARALLEL: FALSE
+          QISKIT_IGNORE_USER_SETTINGS: TRUE
+          RUST_BACKTRACE: 1
+  test-rust:
+    name: ${{ inputs.runner }}-tests-Rust-${{ inputs.python-version }}
+    runs-on: ${{ inputs.runner }}
+    env:
+      QISKIT_SUPPRESS_PACKAGING_WARNINGS: Y
+      PIP_CACHE_DIR: ${{ github.workspace }}/.pip
+      QISKIT_TEST_CAPTURE_STREAMS: 1
+      HAVE_VISUAL_TESTS_RUN: false
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@1.79
+        if: inputs.python-version == '3.9'
```

Comment:
Is this needed if we have the `rustup override stable` manually called below?

Reply:
This is only done when building from sdist, I believe we override to make sure our sdists work correctly with the latest stable version of rust.